test/test.js

```diff
@@ -732,6 +732,52 @@ test('CircuitBreaker semaphore rate limiting', (t) => {
   });
 });
 
+test('CircuitBreaker with failureCheck returns false', (t) => {
+  t.plan(3);
+  const breaker = cb(passFail, { errorThresholdPercentage: 10 });
+
+  breaker.failureCheck((e) => {
+    t.equals(e, 'Error: -1 is < 0', 'passed checked error');
+    return false;
+  });
+
+  breaker.fire(-1)
+    .then(() => t.fail('Tested function should throw an error'))
+    .catch((e) => t.equals(e, 'Error: -1 is < 0', 'caught expected error'))
+    .then(() => t.notOk(breaker.opened, 'Breaker should not open'))
+    .catch(t.end);
+});
+
+test('CircuitBreaker with failureCheck returns true', (t) => {
+  t.plan(3);
+  const breaker = cb(passFail, { errorThresholdPercentage: 10 });
+
+  breaker.failureCheck((e) => {
+    t.equals(e, 'Error: -1 is < 0', 'passed checked error');
+    return true;
+  });
+
+  breaker.fire(-1)
+    .then(() => t.fail('Tested function should throw an error'))
```

Comment:
It was only when I got to this test that I realized what's really happening here. In all cases, even if the error is determined to be ok, the circuit fails. This feature only prevents the statistic from being registered, but doesn't have an effect on the user code flow. 

I'm not sure about this. 

I think perhaps that `circuit.fire()` should return a resolved promise if the failure check function returns true. This is similar to how the `fallback` feature works. In the event of a failure, if a fallback function is available, the circuit resolves successfully. See: https://github.com/bucharest-gold/opossum/blob/master/test/test.js#L263-L266